tls/s2n_client_hello.c

```diff
@@ -308,13 +342,20 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     /* Find potential certificate matches before we choose the cipher. */
     GUARD(s2n_conn_find_name_matching_certs(conn));
 
-
     /* Now choose the ciphers and the cert chain. */
     GUARD(s2n_set_cipher_and_cert_as_tls_server(conn, client_hello->cipher_suites.data, client_hello->cipher_suites.size / 2));
 
     /* And set the signature and hash algorithm used for key exchange signatures */
-    GUARD(s2n_choose_sig_scheme_from_peer_preference_list(conn, &conn->handshake_params.client_sig_hash_algs,
-                                                           &conn->secure.conn_sig_scheme));
+    if (conn->actual_protocol_version < S2N_TLS13) {
+        GUARD(s2n_choose_sig_scheme_from_peer_preference_list(conn,
+            &conn->handshake_params.client_sig_hash_algs,
+            &conn->secure.conn_sig_scheme));
+    } else {
+        /* in the case of TLS 1.3, signature algo and cert selection uses a different approach */
+        GUARD(s2n_choose_tls13_sig_scheme_and_set_cert(conn,
```

Comment:
I would take your last suggestion - I don't think it's a good idea and would be more tricky if we tried to unifying 1.2 and 1.3 paths since there's some fundamental differences.